receiver/receivertest/contract_checker.go

```diff
@@ -0,0 +1,437 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package receivertest // import "go.opentelemetry.io/collector/receiver/receivertest"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/receiver"
+)
+
+// UniqueIDAttrName is the attribute name that is used in log records/spans/datapoints as the unique identifier.
+const UniqueIDAttrName = "test_id"
+
+// UniqueIDAttrVal is the value type of the UniqueIDAttrName.
+type UniqueIDAttrVal = string
```

Comment:
Should we apply the type definition to enforce it? Otherwise just `string` can be used instead

```suggestion
type UniqueIDAttrVal string
```

Reply:
I went back and forth on this, chose an alias to have less typing in the Generator, but no strong opinion. If you feel a stronger type could help we can change to that.

Reply:
Changed.